x11-libs/cairo/cairo-1.14.12.recipe

```diff
@@ -0,0 +1,106 @@
+SUMMARY="Multi-platform 2D graphics library"
+DESCRIPTION="Cairo is a 2D graphics library with support for multiple output \
+devices. Currently supported output targets include the X Window \
+System (via both Xlib and XCB), quartz, win32, and image buffers, \
+as well as PDF, PostScript, and SVG file output. Experimental backends \
+include OpenGL, BeOS, OS/2, and DirectFB."
+HOMEPAGE="http://cairographics.org/"
+COPYRIGHT="2000, 2002, 2004-2007 Keith Packard
+	2002-2003 University of Southern California
+	2004-2010 Red Hat, Inc.
+	2005-2010 Mozilla Corporation
+	2006-2009 Adrian Johnson
+	2007-2009 Chris Wilson
+	2006-2013 Intel Corporation
+	2011 Andrea Canciani
+	2011 Samsung Electronics
+	2010-2011 Linaro Limited
+	2009-2010 Eric Anholt
+	2002-2010 many others"
+LICENSE="GNU LGPL v2.1
+	MPL v1.1"
+REVISION="1"
+SOURCE_URI="http://cairographics.org/releases/cairo-$portVersion.tar.xz"
+CHECKSUM_SHA256="8c90f00c500b2299c0a323dd9beead2a00353752b2092ead558139bd67f7bf16"
+PATCHES="cairo-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+libVersion="2.11400.12"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	cairo$secondaryArchSuffix = $portVersion compat >= 1
+	lib:libcairo$secondaryArchSuffix = $libVersionCompat
+	lib:libcairo_script_interpreter$secondaryArchSuffix = $libVersionCompat
+	lib:libcairo_gobject$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libfontconfig$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libglib_2.0$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:libpixman_1$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	cairo${secondaryArchSuffix}_devel = $portVersion compat >= 1
+	devel:libcairo$secondaryArchSuffix = $libVersionCompat
+	devel:libcairo_script_interpreter$secondaryArchSuffix = $libVersionCompat
+	devel:libcairo_gobject$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	cairo$secondaryArchSuffix == $portVersion base
+	devel:libfontconfig$secondaryArchSuffix
+	devel:libpixman_1$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libfontconfig$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:libglib_2.0$secondaryArchSuffix
+	devel:libpixman_1$secondaryArchSuffix
+	devel:libpng16$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:automake
```

Comment:
cmd:autorecon is part of the package called by cmd:autoconf ... (automake shouldn't be in here also) ...

Reply:
Yep, but if the recipe calls `autoreconf` then it is recommended to mention it in the `BUILD_PREREQUIRES`.
This is safer in case autoreconf is moved to a subpackage or if, at some point in the future, an alternate package provides a `cmd:autoreconf`.

Reply:
This doesn't add up to the comments on https://github.com/haikuports/haikuports/pull/2230 ?

Reply:
So we should drop "autoconf, automake" here? Not sure you "can" remove aclocal ...

Reply:
Generally, you could just drop anything that's not called by the recipe. `autoconf`, `automake` are dependencies of `autoreconf`, so they would be pulled automatically.

Reply:
I think it is better to keep in `BUILD_PREREQUIRES` most of these `cmd:*`  which are mentioned in configure.
Although `cmd:autoreconf` is (currently) provided by the same package which also provides `cmd:aclocal`, `cmd:autoconf` and `cmd:automake`, this could change in the future, so we should not, imho, drop these.

`cmd:makeinfo` can be dropped because it is not usefull for us. The only thing it does, if I'm not mistaken, is that it creates a `dir` file in `$infoDir`, and we don't like to have this `dir` file.

Reply:
I did some changes back, but got told to add/remove them, getting you guys in line would be nice for us to know if we should use them or not :)

Reply:
To decide whether or not it would be OK to remove a `cmd:` from `BUILD_PREREQUIRES`, we need to know if the recipe would still build if we did remove that `cmd:` and also had **_each_** of the remaining `cmd:` items in a **_separate_** package or subpackage.
We should not assume that any set of commands will always be provided by the same package.
There is only one exception, and it's for the trivial commands like `cp`, `mv`, `rm`, `ln`, `mkdir`, `install` (and probably other very common commands), which are in coreutils.
BTW, commands like `tr`, `sed`, and maybe `cut` too, should be mentioned in `BUILD_PREREQUIRES` if used in the recipe or by the build processes.
In other words, it would only be OK to not mention a `cmd:` if it is absolutely obvious that it will allways be in coreutils.

Reply:
> Generally, you could just drop anything that's not called by the recipe. `autoconf`, `automake` are dependencies of `autoreconf`, so they would be pulled automatically.

@alaviss: OK, I see what you mean. I have to admit that you are right.
But I still think that when we know that the configure script tests for the presence of aclocal, autoconf, autoheader, automake, then it is safer to mention the corresponding `cmd:` in `BUILD_PREREQUIRES`.

Reply:
If I'm not mistaken, I already saw recipes which only call runConfigure and make, and which:
* fail to build if we don't mention at least some of these `cmd:`.
* build fine if we do mention at least a minimal set of `cmd:`.

Consider `cmd:pkg_config`. It is never called directly from the recipe. If we chose to only mention those `cmd:` that we call from the recipe, then we would "have" to drop `cmd:pkg_config`. We don't do that because we know that the build process needs it.

So here, if we know that some tools are needed (or even merely tested for their presence) during the build, then we should mention them in `BUILD_PREREQUIRES`.

@alaviss (and others), what do you think?
Would you agree to keep the `cmd:aclocal`, `cmd:autoconf`, `cmd:autoheader` and `cmd:automake` even if the recipe only calls autoreconf?

Reply:
When haikuporter sees the `cmd:autoreconf` in `BUILD_PREREQUIRES`, it searches for a package providing it, and "activates" it in the chroot. Now, since it is currently provided by the autoconf package, the chroot will automatically also have all the other `cmd:` provided by that package: `cmd:autoconf`, `cmd:autoheader`, `cmd:autom4te`, `cmd:autoscan`, `cmd:autoupdate` and `cmd:ifnames`.
So, even if you omit these commands, they will be available in the chroot if you mention `cmd:autoreconf`. And you might not even notice that some of them are called.
The problem is that if, at some point in the future, the layout of autoconf package is changed and some commands moved to subpackages or removed from that package, then they would no longer be "pulled" by `cmd:autoreconf`.
That's why, imho, it is safer to _also_ mention `cmd:autoconf` and `cmd:autoheader`, instead of just `cmd:aclocal`, `cmd:automake` and `cmd:autoreconf`.